zhaquirks/tuya/ts0601_trv_sas.py

```diff
@@ -81,6 +85,25 @@ def _update_attribute(self, attrid, value):
             self.endpoint.device.thermostat_bus.listener_event(
                 LOCAL_TEMP_REPORTED, value
             )
+        elif attrid == SCHEDULE_ENABLED_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                SCHEDULE_STATE_REPORTED, value
+            )
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Implement writeable attributes."""
+
+        if "system_mode" in attributes:
+
+            async def _disable_schedule():
+                await asyncio.sleep(3)
```

Comment:
I think that Z2M adds a delay because there is 2 writes. Have you tried without the `sleep`?.

Reply:
I think they added it to work-around some race condition in device firmware. I've tried to reproduce this condition but I've failed to. For now I'll remove this delay.